evm_loader/lib/src/solana_simulator/utils.rs

```diff
@@ -0,0 +1,192 @@
+        genesis_config,
+        mint_keypair,
+        voting_keypair,
+        validator_pubkey,
+    })
+}
+
+pub async fn deactivated_features(rpc: &impl Rpc) -> Result<Vec<Pubkey>, Error> {
+    let feature_keys: Vec<Pubkey> = solana_sdk::feature_set::FEATURE_NAMES
+        .keys()
+        .copied()
+        .collect();
+    let features = rpc.get_multiple_accounts(&feature_keys).await?;
+
+    let mut result = Vec::with_capacity(feature_keys.len());
+    for (pubkey, feature) in feature_keys.iter().zip(features) {
+        let is_activated = feature
+            .and_then(|a| solana_sdk::feature::from_account(&a))
+            .and_then(|f| f.activated_at)
+            .is_some();
+
+        if !is_activated {
+            result.push(*pubkey);
+        }
+    }
+
+    Ok(result)
+}
+
+pub async fn sync_sysvar_accounts(rpc: &impl Rpc, bank: &mut Bank) -> Result<(), Error> {
+    let keys = sysvar::ALL_IDS.clone();
+    let accounts = rpc.get_multiple_accounts(&keys).await?;
+
+    for (key, account) in keys.into_iter().zip(accounts) {
+        let Some(account) = account else {
+            continue;
+        };
+
+        match key {
+            sysvar::clock::ID => {
+                use sysvar::clock::Clock;
+
+                let clock: Clock = bincode::deserialize(&account.data)?;
+                bank.set_sysvar_for_tests(&clock);
+            }
+            sysvar::epoch_schedule::ID => {
+                use sysvar::epoch_schedule::EpochSchedule;
+
+                let epoch_schedule: EpochSchedule = bincode::deserialize(&account.data)?;
+                bank.set_sysvar_for_tests(&epoch_schedule);
+            }
+            sysvar::rent::ID => {
+                use sysvar::rent::Rent;
+
+                let rent: Rent = bincode::deserialize(&account.data)?;
+                bank.set_sysvar_for_tests(&rent);
+            }
+            sysvar::rewards::ID => {
+                use sysvar::rewards::Rewards;
+
+                let rewards: Rewards = bincode::deserialize(&account.data)?;
+                bank.set_sysvar_for_tests(&rewards);
+            }
+            sysvar::slot_hashes::ID => {
+                use sysvar::slot_hashes::SlotHashes;
+
+                let slot_hashes: SlotHashes = bincode::deserialize(&account.data)?;
+                bank.set_sysvar_for_tests(&slot_hashes);
+            }
+            sysvar::slot_history::ID => {
+                use sysvar::slot_history::SlotHistory;
+
+                let slot_history: SlotHistory = bincode::deserialize(&account.data)?;
+                bank.set_sysvar_for_tests(&slot_history);
+            }
+            sysvar::stake_history::ID => {
+                use sysvar::stake_history::StakeHistory;
+
+                let stake_history: StakeHistory = bincode::deserialize(&account.data)?;
+                bank.set_sysvar_for_tests(&stake_history);
+            }
+            #[allow(deprecated)]
+            id if sysvar::fees::check_id(&id) => {
```

Comment:
Why do you use `...::check_id(&id)` in some cases instead of `ID`?

Reply:
`ID` does not exists for deprecated sysvars